cassandra-executor/src/main/java/io/mesosphere/mesos/frameworks/cassandra/jmx/NodeCleanupJob.java

```diff
@@ -13,51 +13,73 @@
  */
 package io.mesosphere.mesos.frameworks.cassandra.jmx;
 
+import io.mesosphere.mesos.frameworks.cassandra.CassandraFrameworkProtos;
 import org.apache.cassandra.db.compaction.CompactionManager;
+import org.apache.mesos.Protos;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import java.util.HashMap;
-import java.util.Map;
+import java.util.List;
 
-public class NodeCleanupJob extends AbstractKeyspacesJob {
+public class NodeCleanupJob extends AbstractNodeJob {
     private static final Logger LOGGER = LoggerFactory.getLogger(NodeCleanupJob.class);
 
-    private final Map<String, CompactionManager.AllSSTableOpStatus> keyspaceStatus = new HashMap<>();
+    public NodeCleanupJob(Protos.TaskID taskId) {
+        super(taskId);
+    }
 
-    public NodeCleanupJob() {
+    @Override
+    public CassandraFrameworkProtos.ClusterJobType getType() {
+        return CassandraFrameworkProtos.ClusterJobType.CLEANUP;
     }
 
     public boolean start(JmxConnect jmxConnect) {
         if (!super.start(jmxConnect))
             return false;
 
-        LOGGER.info("Initiated cleanup job for keyspaces {}", keyspaces);
+        LOGGER.info("Initiated cleanup job for keyspaces {}", getRemainingKeyspaces());
 
         return true;
     }
 
-    public void cleanupBlocking() {
-        for (String keyspace : keyspaces) {
-            int statusOrdinal = 0;
-            try {
-                statusOrdinal = jmxConnect.getStorageServiceProxy().forceKeyspaceCleanup(keyspace);
-            } catch (Exception e) {
-                LOGGER.error("Cleanup for keyspace " + keyspace + " failed", e);
-            }
-            CompactionManager.AllSSTableOpStatus status = CompactionManager.AllSSTableOpStatus.values()[statusOrdinal];
-            keyspaceStatus.put(keyspace, status);
-            LOGGER.info("Cleanup for keyspace {} finished with status {}", keyspace, status);
-        }
-        finished();
-    }
+    @Override
+    public void startNextKeyspace() {
+        final String keyspace = super.nextKeyspace();
+        if (keyspace == null)
+            return;
 
-    private void finished() {
+        // cleanup is a blocking operation that may take very long
+        new Thread("Cleanup " + keyspace) {
```

Comment:
please move this to being a runnable that is submitted to an executor service, then track the returned future.